### ISSUE TITLE
Multiple reboots 

### ISSUE THREAD

**Issue by deco9999** [RikkaApps/Riru#62]
Device: Samsung s7 stock oreo rooted with  magisk
Everything works fine except when i restart the phone it restars multiple times (usually 3 times),
Even after disabling edxposed and restarting the problem persisted so i'm pretty sure Riru-core is the cause.

**Comment by RonanTetsu** [comment]
Similar issue, on oneplus 7 pro, but only when using mods.. (Swift installer is my primary cause for this glitch, that I do not know why.)

**Comment by NHellFire** [comment]
Also seeing this on my Note 9 with just core enabled. Will try and get a logcat when I have the time to wait for a crash

**Comment by NHellFire** [comment]
Tested again on my Note 9. It appears to be causing an exception which results in a panic. Last thing in last_kmsg was:
```
<1>[ 9942.410253]  [4:    highpool[3]:10700] =============uH fault handler logging=============
<1>[ 9942.410270]  [4:    highpool[3]:10700] [System registers]
<1>[ 9942.410280]  [4:    highpool[3]:10700] ESR_EL2: 5a000000	HCR_EL2: 80000003	HPFAR_EL2: 80029f000
```

https://github.com/exynos-linux-stable/crownlte/blob/e26851fcc089e46254db6c859c7bd55624a2cce2/init/uh_fault_handler.c#L95-L122
Looks like that's part of KNOX. I'll try a different kernel and report back.

**Comment by TitusLabienus** [comment]
I also have reboots every couple of hours on my Note 9.
From trying around by enabling/disabling Magisk modules, it too seems to me that enabling Riru Core (or possibly EdXposed) causes the issue, even when no Xposed modules are enabled.

**Comment by NHellFire** [comment]
I've not had any reboots on my Note 9 on a different kernel with only Riru Core enabled.
@TitusLabienus if you're using a kernel without KNOX, it's likely EdXposed causing it

**Comment by TitusLabienus** [comment]
@NHellFire Thank you for the info. I use the default kernel from the Samsung firmware. Ok, I'll see if it stops rebooting when only Riru Core is enabled.

**Comment by TitusLabienus** [comment]
@NHellFire Does the standard EdXposed work on your device? I only get it work with  a special version. (https://forum.xda-developers.com/galaxy-note-9/themes/edxposed-android-10-q-exynos-t4061113) So yes, maybe this special version causes that restart bug for me.

**Comment by TitusLabienus** [comment]
I still have this issue with Riru - Core v21.3 and Magisk v21.0.

Steps to reproduce:
- Enable Riru - Core (Reproducible also when other Riru modules are disabled
- Notice Reboot every few hours

Device: Galaxy Note 9, Android 10

**Comment by TitusLabienus** [comment]
@RikkaW Can you reproduce this issue, or do you need any logs etc. for doing so?

Hopefully this issue can be fixed, it prevents me from using any Riro / Exposed modules.

**Comment by RikkaW** [comment]
@TitusLabienus Your problem is highly possible caused by EdXposed rather than Riru itself. Riru itself is super simple, if the can boot means no problem at all.

**Comment by RikkaW** [comment]
Close this issue for Riru v22 changes the inject method.

**Comment by TitusLabienus** [comment]
@RikkaW Thanks for your reply. I have the issue also if I disable the EdXposed module and only enable the Core one. As soon as I disable the Core module, it does not reboot anymore. So it cannot be EdXposed. Could you reopen the ticket please?

**Comment by RikkaW** [comment]
@TitusLabienus Have you tried [v22.0](https://github.com/RikkaApps/Riru/releases/tag/v22.0)?

**Comment by TitusLabienus** [comment]
@RikkaW No, I have only tried the last stable version (v21.3). I will try with v22.0 ( and only the Core module enabled) and report back if the reboots still occur.

**Comment by TitusLabienus** [comment]
@RikkaW Using only the Core module, I had no reboot for several days (whereas I had reboots every few hours with v21.3). So far the reboots seem to be fixed in v22.0. Thanks!
